core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,515 @@
+              visibleElements.put((DAnnotation) nodeTarget, aNode);
+            }
+          }
+        }
+      }
+    }
+    for (Entry<DAnnotation, DDiagramElement> me : visibleElements.entrySet()) {
+      if (!selectedTitleBlocks.contains(me.getKey())) {
+        EObject container = me.getValue().eContainer();
+        if (container instanceof DSemanticDiagram) {
+          DAnnotation annotation = ((DAnnotation) (me.getValue().getTarget()));
+          if (isDiagram) {
+            if (isDiagramTitleBlock(annotation)) {
+              annotation.getDetails().put(VISIBILITY, FALSE);
+              DiagramServices.getDiagramServices().removeAbstractDNodeView((DNodeContainer) me.getValue());
+            }
+          } else {
+            if (!isDiagramTitleBlock(annotation)) {
+              annotation.getDetails().put(VISIBILITY, FALSE);
+              DiagramServices.getDiagramServices().removeAbstractDNodeView((DNodeContainer) me.getValue());
+            }
+          }
+        }
+      }
+    }
+    for (DAnnotation aTitleBlock : selectedTitleBlocks) {
+      if (!visibleElements.containsKey(aTitleBlock)) {
+
+        createTitleBlockView(context, aTitleBlock, diagram);
+      }
+    }
+    return context;
+  }
+
+  private AbstractDNode createTitleBlockView(EObject context, DAnnotation titleBlock, DDiagram diagram) {
+    String mappingName = IMappingNameConstants.CDB_TITLE_BLOCK;
+    NodeMapping mapping = DiagramServices.getDiagramServices().getNodeMapping(diagram, mappingName);
+    if (context instanceof DNodeList) {
+      if (null != titleBlock.getDetails().get(VISIBILITY)) {
+        titleBlock.getDetails().put(VISIBILITY, TRUE);
+      }
+      return DiagramServices.getDiagramServices().createDNodeListElement(mapping, titleBlock,
+          (DragAndDropTarget) context, diagram);
+    }
+    if (null != titleBlock.getDetails().get(VISIBILITY)) {
+      titleBlock.getDetails().put(VISIBILITY, TRUE);
+    }
+    return DiagramServices.getDiagramServices().createNode(mapping, titleBlock, (DragAndDropTarget) context, diagram);
+  }
+
+  public boolean isUniqueDiagramTitleBlock(EObject diagram) {
+    Collection<DAnnotation> result = new ArrayList<>();
+    if ((diagram instanceof DRepresentation)) {
+      DRepresentation representation = (DRepresentation) diagram;
+      result = representation.getEAnnotations().stream().filter(x -> x.getSource().equals(TITLE_BLOCK))
+          .filter(x -> x.getReferences().size() == 2).collect(Collectors.toList());
+    }
+    return (result.size() == 0);
+  }
+
+  public boolean isUniqueElementTitleBlock(Object elementView, EObject diagram) {
+    Collection<DAnnotation> result = new ArrayList<>();
+    if ((diagram instanceof DRepresentation)) {
+      DRepresentation representation = (DRepresentation) diagram;
+      result = representation.getEAnnotations().stream().filter(x -> x.getSource().equals(TITLE_BLOCK))

```

Comment:
instead of TITLE_BLOCK use ELEMENT_TITLE_BLOCK.

Reply:
Done.